### PR TITLE
upgrade typescript, types, node and webpack for typescript template

### PR DESCRIPTION
**Pull request by mlecoq**
## What did you implement:

I have updated template for typescript in order to use node 8, to have types for streams (missing in previous version) and to use webpack 4

## How did you implement it:
I have updated release in package.json, modified node version in serverless.yml and modify webpack conf (rules) to support webpack 4

## How can we verify it:
`~/bin/serverless create -t aws-nodejs-typescript`
`npm or yarn install`
`sls deploy`

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/create/templates/aws-nodejs-typescript/webpack.config.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ Object.keys(slsw.lib.entries).forEach(key => (
 ));
 
 module.exports = {
+  mode: 'production',
```
Comment (HyperBrain) [inline lib/plugins/create/templates/aws-nodejs-typescript/webpack.config.js]: This should be `slsw.lib.webpack.isLocal ? 'development' : 'production'` to enable debugging features when running locally.

**Comment by mlecoq** [comment]
Done !

**Comment by sime** [comment]
Would it be possible to compile the JS to ES6?

I just try to worked it out myself, but wasn't able to figure it.

**Comment by HyperBrain** [comment]
@sime According to the TS docs, you should set `target: ES6` in your TypeScript config.

**Comment by HyperBrain** [comment]
@mlecoq regarding my last comment here, do you think the target should be explicitly set to code compatible with Node 8 as we now set it as AWS target here? Imo it does not make sense to compile to code for older versions.

**Comment by mlecoq** [comment]
You're right, I will update that

**Comment by sime** [comment]
@mlecoq That commit does not actually compile to ES6.

You need to set target:

```json
{
  "compilerOptions": {
    "sourceMap": true,
    "target": "es6"
  },
  "exclude": [
    "node_modules"
  ]
}
```

The above will compile the `handle.ts` file to ES6. 

Though I recommend the following:

```json
{
  "compilerOptions": {
    "sourceMap": true,
    "target": "es6",
    "lib": [
      "esnext"
    ],
    "moduleResolution": "node"
  },
  "exclude": [
    "node_modules"
  ]
}
```

As a developer starts doing work, `esnext` lib will be required, and `moduleResolution` needs to be set to `node`.

I have a very simple function, and I need the above to make it compile.

**Comment by mlecoq** [comment]
@sime - tsconfig is fixed

**Comment by HyperBrain** [comment]
@mlecoq @sime Thanks for finalizing. Merging now.

**Pull request merged**